sns-kfh-transform/main.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/honeycombio/agentless-integrations-for-aws/common"
+	"github.com/sirupsen/logrus"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+)
+
+type snsEntityForHny struct {
+	events.SNSEntity
+	Message map[string]interface{}
+}
+
+func handler(ctx context.Context, input events.KinesisFirehoseEvent) (events.KinesisFirehoseResponse, error) {
+	var response events.KinesisFirehoseResponse
+	for _, record := range input.Records {
+		var snsEntity events.SNSEntity
+		err := json.Unmarshal(record.Data, &snsEntity)
+		if err != nil {
+			logrus.WithError(err).Warn("failed unmarshal outer event")
```

Comment:
I think we should add a `continue` here if there's an error, so we don't try to work with an event we couldn't parse further down.

Also, I see that in common/kinesis.go we're sticking events we couldn't process back onto the stream along with `failedRecord.Result = events.KinesisFirehoseTransformedStateProcessingFailed`. Should we be doing that here too?

(this comment applies for all the spots there's have error handling in this function)

Reply:
Good call, done.